### PR TITLE
Use `CodeBlock` to generate human-readable code in JIT

### PR DESCRIPTION
**Pull request by asi1024**
This PR fixes JIT code with `CodeBlock` object, which we implemented in CuPy fusion to generate human-readable CUDA code.

### CODE REVIEW

File: cupyx/jit/_types.py [hunk 1]
```diff
@@ -43,4 +43,4 @@ def __str__(self):
         if dtype == numpy.float16:
             # For the performance
             dtype = numpy.dtype('float32')
-        return _typenames[dtype]
+        return dtype_to_ctype[dtype]
```
Comment (leofang) [inline cupyx/jit/_types.py:46]: Looks like this can be replaced by `cupy.core._scalar.get_typename()`?

Comment (asi1024) [reply cupyx/jit/_types.py:46]: It is a good idea! I will do it on this PR.

**Comment by asi1024** [comment]
Rebased.

**Review by emcastillo — APPROVED** [review]

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by asi1024** [comment]
This CI failure will be fixed in https://github.com/cupy/cupy/pull/4648.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Pull request merged**